src/components/Zap/ZapAction/Zapwidget.tsx

```diff
@@ -0,0 +1,268 @@
+import {
+  refuelAllowChains,
+  themeAllowChains,
+} from 'src/components/Widgets/Widget.types';
+import { tokens } from 'src/config/tokens';
+import { publicRPCList } from 'src/const/rpcList';
+import { ThemesMap } from 'src/const/themesMap';
+import {
+  TrackingAction,
+  TrackingCategory,
+  TrackingEventParameter,
+} from 'src/const/trackingKeys';
+import { useMemelist } from 'src/hooks/useMemelist';
+import { useWelcomeScreen } from 'src/hooks/useWelcomeScreen';
+import { useUserTracking } from 'src/hooks/userTracking';
+import { useActiveTabStore } from 'src/stores/activeTab';
+import { isIframeEnvironment } from 'src/utils/iframe';
+import { useConfig } from 'wagmi';
+
+export function ZapWidget({
+  starterVariant,
+  fromChain,
+  fromToken,
+  toChain,
+  toToken,
+  fromAmount,
+  allowChains,
+  widgetIntegrator,
+  activeTheme,
+  autoHeight,
+}: WidgetProps) {
+  const [widgetTheme, configTheme] = useThemeStore((state) => [
+    state.widgetTheme,
+    state.configTheme,
+  ]);
+  const formRef = useRef<FormState>(null);
+  const { i18n } = useTranslation();
+  const { trackEvent } = useUserTracking();
+  const wagmiConfig = useConfig();
+  const { isMultisigSigner, getMultisigWidgetConfig } = useMultisig();
+  const { multisigWidget, multisigSdkConfig } = getMultisigWidgetConfig();
+  const { activeTab } = useActiveTabStore();
+  const partnerName = configTheme?.uid ?? 'default';
+  const { tokens: memeListTokens } = useMemelist({
+    enabled: partnerName === ThemesMap.Memecoins,
+  });
+  const { openWalletMenu } = useWalletMenu();
+  const widgetCache = useWidgetCacheStore((state) => state);
+
+  const router = useRouter();
+  useEffect(() => {
+    router.prefetch('/', { kind: PrefetchKind.FULL });
+    router.prefetch('/gas', { kind: PrefetchKind.FULL });
+  }, [router]);
+
+  const { welcomeScreenClosed, enabled } = useWelcomeScreen(activeTheme);
+
+  const isGasVariant = activeTab === TabsMap.Refuel.index;
+  const allowedChainsByVariant = useMemo(
+    () =>
+      starterVariant === TabsMap.Refuel.variant
+        ? refuelAllowChains
+        : partnerName === ThemesMap.Memecoins
+          ? themeAllowChains
+          : [],
+    [starterVariant, partnerName],
+  );
+
+  const integratorStringByType = useMemo(() => {
+    if (widgetIntegrator) {
+      return widgetIntegrator;
+    }
+    // all the trafic from web on "/gas"
+    if (isGasVariant) {
+      return process.env.NEXT_PUBLIC_WIDGET_INTEGRATOR_REFUEL;
+    }
+
+    return process.env.NEXT_PUBLIC_WIDGET_INTEGRATOR;
+  }, [widgetIntegrator, isGasVariant]) as string;
+
+  const overriddenTheme = {
+    ...widgetTheme.config.theme,
+    container: {
+      ...widgetTheme.config.theme?.container,
+      boxShadow: 'unset',
+      '@media (min-width:600px)': {
+        // @ts-ignore or @ts-expect-error
+        ...widgetTheme.config.theme?.container['@media (min-width:600px)'],
+        boxShadow: 'unset',
+      },
+    },
+  };
```

Comment:
This is the part where I am overriden the boxShadow. It returned a ts-error which I am ignoring by using `// @ts-ignore or @ts-expect-error`. Wdyt?

Reply:
It´s deprecated! Now, that magic happens within "useWidgetConfig"